src/parser.rs

```diff
@@ -952,6 +953,29 @@ impl Library {
             "attribute" => parser.ignore_element(),
             _ => Err(parser.unexpected_element(elem)),
         })?;
+        let mut fix_pos = 0;
+        // Fixing the closure attributes in case they're badly set.
+        for index in 0..params.len() {
+            if params[index].instance_parameter {
+                fix_pos += 1;
+            } else if let Some(pos) = params[index].closure {
+                if params[pos - fix_pos].closure.is_none() {
+                    params[pos - fix_pos].closure = Some(pos - fix_pos);
+                }
+            }
+        }
+        // Adding closure attribute for callback parameters in case there is something completely
+        // wrong.
+        if is_callback && params.last().map(|x| x.closure.is_none()).unwrap_or(false) {
```

Comment:
The attribute is not supposed to be on the callback types, at all. The last parameter is always going to be the one we're looking for here

Reply:
It's how `gir` knows unfortunately. We use this attribute everywhere.